### PR TITLE
[Reservation] Adding new commands(calculateRefund/Return/Archive/Unarchive)

### PR DESCRIPTION
**Pull request by gaoyp830**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by gaoyp830** [comment]
Swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-03-01/reservations.json

File: src/Reservations/examples/Invoke-AzReservationCalculateRefund.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                                                             }
+PolicyError                                                : {}
+PricingRefundAmount                                        : {
+                                                               "currencyCode": "USD",
+                                                               "amount": 0.75
+                                                             }
+Quantity                                                   : 1
+ResourceGroupName                                          :
+SessionId                                                  : b0a96155-5f75-4138-b01f-443130f5516e
+```
```
Comment (wyunchi-ms) [inline src/Reservations/examples/Invoke-AzReservationCalculateRefund.md:47]: Please add description

File: src/Reservations/examples/Invoke-AzReservationReturn.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                                                             }
+PolicyError                                                : {}
+PricingRefundAmount                                        : {
+                                                               "currencyCode": "USD",
+                                                               "amount": 2.58
+                                                             }
+Quantity                                                   : 1
+ResourceGroupName                                          :
+SessionId                                                  : 93fe5df2-d888-47c5-b00c-cd0ccb1f29b9
+```
```
Comment (wyunchi-ms) [inline src/Reservations/examples/Invoke-AzReservationReturn.md:48]: Please add description

Comment (gaoyp830) [reply src/Reservations/examples/Invoke-AzReservationReturn.md]: Updated with description

File: src/Reservations/examples/Invoke-AzReservationCalculateRefund.md (hunk 1 above)
Comment (gaoyp830) [reply src/Reservations/examples/Invoke-AzReservationCalculateRefund.md:47]: Updated with description

**Review by wyunchi-ms — APPROVED** [review]

**Comment by wyunchi-ms** [comment]
Please update the changelog at this [breanch](https://github.com/Azure/azure-powershell/pull/19569) @gaoyp830

**Pull request merged**